src/main/java/com/zenjava/javafx/maven/plugin/NativeMojo.java

```diff
@@ -185,6 +185,22 @@
      */
     protected String appName;
 
+    /**
+     * Set this to true if you would like to install your application in the system directory (i.e. Program Files on
+     * Windows) instead of a user specific one. Note that when you activate this option, you need admin privilege to
+     * install your application.
+     *
+     * @parameter default-value=false
+     */
+    protected boolean installForAllUsers;
+
+    /**
+     * The icon file to customize icon application launcher. On Windows, the .ico format is expected, and on Mac it is
+     * .icns. No icon is embedded into launcher on Linux.
+     *
+     * @parameter
+     */
+    protected File appIcon;

```

Comment:
This is already possible by putting that icon into system-specific folder ```src/main/deploy/package/windows/{your-app-name}.ico``` because its managed by the packager. I don't see the need for this, its just documentation whats missing IMHO.